File: src/app/admin/XXXXX.Admin.UI/src/services/translationService.ts
import axios from "axios";

import { Translation, TranslationDTO } from "@/domain/models";
import { ITranslationService } from "@/interfaces";

import { TRANSLATIONS_URL } from "@/config";

export class TranslationService implements ITranslationService {
    type: string = "TranslationService";

    async getMany(): Promise<Translation[]> {
        const response = await axios.get(TRANSLATIONS_URL);
        const dtos: TranslationDTO[] = response.data;

        const results = dtos.map(d => new Translation(d));
        return results;
    }
}

File: src/app/admin/XXXXX.Admin.Core/Services/TranslationService.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

using AutoMapper;

using Bones.Flow;

using XXXXX.Domain.Models;
using XXXXX.Admin.Core.Interfaces;
using XXXXX.Admin.Core.ViewModels;
using XXXXX.Admin.Core.Requests;

namespace XXXXX.Admin.Core.Services
{
    public class TranslationService : ITranslationService
    {
        private IQueryHandler<TranslationsQuery, IEnumerable<Translation>> _translationsQueryHandler;
        private IMapper _mapper;

        public TranslationService(
            IQueryHandler<TranslationsQuery, IEnumerable<Translation>> translationsQueryHandler,
            IMapper mapper
        )
        {
            _translationsQueryHandler = translationsQueryHandler;
            _mapper = mapper;
        }

        public async Task<IEnumerable<TranslationViewModel>> GetMany(RequestHeaders headers)
        {
            var query = new TranslationsQuery()
            {
                ApplicationId = headers.ApplicationId,
                ActorId = headers.ActorId
            };

            var result = await _translationsQueryHandler.HandleAsync(query);

            return _mapper.Map<IEnumerable<Translation>, IEnumerable<TranslationViewModel>>(result);
        }
    }
}